tests/lib/BackgroundJob/DummyJobList.php

```diff
@@ -135,6 +140,14 @@ public function setLastRun(IJob $job): void {
 		$job->setLastRun(time());
 	}
 
+	public function hasReservedJob(?string $className = null): bool {
+		return $this->reserved[$className];
+	}
+
+	public function setHasReservedJob(?string $className, bool $hasReserved): void {
+		$this->reserved[$className] = $hasReserved;
+	}
```

Comment:
```suggestion
	public function hasReservedJob(?string $className = null): bool {
		return $this->reserved[$className ?? ''];
	}

	public function setHasReservedJob(?string $className, bool $hasReserved): void {
		$this->reserved[$className ?? ''] = $hasReserved;
	}
```

I did not know null was allowed as array key, it appears to be silently casted to string and so it works, but I would feel better if the cast is explicit.